src/Service/TrustedDevice/Crypto/HaliteCryptoHelper.php

```diff
@@ -0,0 +1,96 @@
+<?php
+
+declare(strict_types = 1);
+
+/**
+ * Copyright 2024 SURFnet bv
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Surfnet\Tiqr\Service\TrustedDevice\Crypto;
+
+use Exception;
+use ParagonIE\Halite\Alerts\InvalidKey;
+use ParagonIE\Halite\Symmetric\Crypto;
+use ParagonIE\Halite\Symmetric\EncryptionKey;
+use ParagonIE\HiddenString\HiddenString;
+use Surfnet\Tiqr\Service\TrustedDevice\Exception\DecryptionFailedException;
+use Surfnet\Tiqr\Service\TrustedDevice\Exception\EncryptionFailedException;
+use Surfnet\Tiqr\Service\TrustedDevice\ValueObject\Configuration;
+use Surfnet\Tiqr\Service\TrustedDevice\ValueObject\CookieValue;
+use Surfnet\Tiqr\Service\TrustedDevice\ValueObject\CookieValueInterface;
+
+class HaliteCryptoHelper implements CryptoHelperInterface
```

Comment:
Did you reuse this from the other StepUp-Gateway encryption solution? If so; this might be a good candidate to move to a shared bundle? This does introduce other complexities, but might be a solution for this code duplication.

Reply:
Discussed: A lot of the copied code is changed, because the purpose of the cookie is very different. 
So the cryptohelper is the most useful part to share. But its a very thin wrapper around Halite, so its more hassle than its worth.